consensus/spos/bls/blsSubroundsFactoryV2.go

```diff
@@ -0,0 +1,65 @@
+package bls
+
+import "github.com/ElrondNetwork/elrond-go/consensus/spos"
+
+// factoryV2 defines the data needed by this factory to create all the subrounds and give them their specific functionality
+type factoryV2 struct {
+	*factory
+}
+
+// NewSubroundsFactoryV2 creates a new factoryV2 object
+func NewSubroundsFactoryV2(factory *factory) (*factoryV2, error) {
```

Comment:
Since we are dealing with factories, my suggestion is to reuse the original factory component and just add a new parameter for the  NewSubroundsFactory function that will switch the `generateBlockSubround` function. In this way we do not need the factoryV2 struct and the GetSubroundsFactoryV2 method that just add duplicated logic.

Reply:
Refactored in PR #4659